### PR TITLE
Add `torch.utils.deterministic.fill_uninitialized_memory` flag

### PR DESCRIPTION
**Pull request by kurtamohler**
Part of #109802

cc @mruberry

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -153,6 +153,8 @@
     "DistributedDataParallelCPU",
     # torch.utils
     "set_module",
+    # torch.utils.deterministic
+    "fill_uninitialized_memory",
```
Comment (albanD) [inline docs/source/conf.py:157]: You can remove this since you added the rst.

**Review by albanD — COMMENTED** [review]
Very nice!
Just small small update for the public API check but SGTM otherwise

**Review by albanD — APPROVED** [review]
Thanks!

**Comment by aaronenyeshi** [comment]
This is causing test/profiler/test_profiler.py: test_tensor_properties to timeout and fail internally. Any chance you can help take a look? Thank you.

https://github.com/pytorch/pytorch/blob/main/test/profiler/test_profiler.py#L2308

**Comment by kurtamohler** [comment]
Thanks for letting me know, I will take a look this week

**Comment by kurtamohler** [comment]
@aaronenyeshi, when I run `python test/profiler/test_profiler.py -k test_tensor_properties` using the changes from this PR, I don't get any timeout. In fact, it seems that none of the tests in `test/profiler/test_profiler.py` even turn deterministic mode on.

I tried adding `torch.use_deterministic_algorithms(True)` to the top of the `test/profiler/test_profiler.py` file after the imports, but that didn't cause any timeouts on my machine either. However, it did cause a few of the other tests in that file (not the `test_tensor_properties` test itself) to fail due to a different pattern of memory allocations in deterministic vs nondeterministic mode.

Are you sure that this PR is really the reason for the internal failure? Maybe it's just a flaky failure. Or maybe the internal testing infrastructure is doing something to turn deterministic mode on before running some of the profiler tests. Could you verify that deterministic mode is actually turned on when the internal CI runs the `test_tensor_properties` test? If the test is not being run with deterministic mode turned on, I'm not sure how this PR could be the cause the of the failure

**Comment by aaronenyeshi** [comment]
Sorry, I think its unrelated, and the automatic bisect flagged this PR. It looks like that test is still having timeout issues. See D50763398 . Please feel free to re-land this diff. Sorry for the inconvenience!

**Review by aaronenyeshi — APPROVED** [review]

**Comment by kurtamohler** [comment]
Huh, for some reason CI did not rerun after I rebased, so the merge command merged this immediately

**Comment by huydhn** [comment]
@pytorchbot drci

(please ignore this comment, I'm testing Dr.CI)

**Pull request closed without merging**